internal/gengapic/genrest.go

```diff
@@ -134,12 +134,14 @@ func (g *generator) restClientUtilities(serv *descriptor.ServiceDescriptorProto,
 	p("")
 	if hasCustomOp {
 		opServName := pbinfo.ReduceServName(opServ.GetName(), g.opts.pkgName)
-		p("opC, err := New%sRESTClient(ctx, opts...)", opServName)
+		p("o := append(opts, option.WithHTTPClient(httpClient))")
```

Comment:
I think what we want to do here is just explicitly pass the httpClient and endpoint and not other user options. This will avoid some validation errors that users might otherwise run into: https://github.com/googleapis/google-api-go-client/blob/33250c025eceebe6d5ed6dae1e0689344fb7ce33/internal/settings.go#L126-L134. WDTY?

Reply:
Ah I see what you're saying. Yeah lets do that.

Reply:
Done.